### ISSUE TITLE
Error executing the spaceAfterTypeColon rule

### ISSUE THREAD

**Issue by Congelli501** [gajus/eslint-plugin-flowtype#43]
Hi, I had the following error (using the eslint bug report template).
Thanks for your hard work on this plug-in !

**What version of ESLint are you using?**
v3.2.2

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**
```javascript
module.exports = {
    "parser": "babel-eslint",
    "env": {
        "es6": true,
        "node": true
    },
    "plugins": [
        "flowtype",
        "json"
    ],
    "parserOptions": {
        "sourceType": "module"
    },
    "settings": {
        "flowtype": {
            "onlyFilesWithFlowAnnotation": true
        }
    },
    "extends": "eslint:recommended",
    "rules": {
        "block-scoped-var": ["error"],
        "consistent-return": ["error"],
        "curly": ["error"],
        "default-case": ["error"],
        "eqeqeq": ["error"],
        "no-alert": ["error"],
        "no-caller": ["error"],
        "no-div-regex": ["error"],
        "no-eval": ["error"],
        "no-extend-native": ["error"],
        "no-extra-bind": ["error"],
        "no-labels": ["error"],
        "no-floating-decimal": ["error"],
        "no-implicit-globals": ["error"],
        "no-implied-eval": ["error"],
        "no-invalid-this": ["error"],
        "no-lone-blocks": ["error"],
        "no-loop-func": ["error"],
        "no-multi-spaces": ["error"],
        "no-multi-str": ["error"],
        "no-new": ["error"],
        "no-new-func": ["error"],
        "no-new-wrappers": ["error"],
        "no-proto": ["error"],
        "no-iterator": ["error"],
        "no-return-assign": ["error"],
        "no-script-url": ["error"],
        "no-self-compare": ["error"],
        "no-throw-literal": ["error"],
        "no-unmodified-loop-condition": ["error"],
        "no-unused-expressions": ["error"],
        "no-useless-call": ["error"],
        "no-useless-concat": ["error"],
        "no-useless-escape": ["error"],
        "no-void": ["error"],
        "yoda": ["error", "never"],

        "strict": ["error"],

        "no-catch-shadow": ["error"],
        "init-declarations": ["error"],
        "no-label-var": ["error"],
        "no-restricted-globals": ["error"],
        "no-shadow": ["error"],
        "no-shadow-restricted-names": ["error"],
        "no-undef-init": ["error"],
        "no-use-before-define": ["error"],

        "no-process-env": ["error"],
        "no-process-exit": ["error"],

        "array-bracket-spacing": ["error"],
        "brace-style": ["error", "1tbs"],
        "camelcase": ["error"],
        "comma-dangle": ["error"],
        "comma-spacing": ["error"],
        "comma-style": ["error"],
        "computed-property-spacing": ["error"],
        "consistent-this": ["error"],
        "eol-last": ["error"],
        "func-names": ["error", "never"],
        "func-style": ["error", "expression"],
        "indent": ["error", "tab"],
        "jsx-quotes": ["error"],
        "key-spacing": ["error"],
        "keyword-spacing": ["error"],
        "linebreak-style": ["error"],
        "max-depth": ["error", {"max": 5}],
        "max-len": ["error", {"code": 120}],
        "max-params": ["error", {"max": 8}],
        "max-statements-per-line": ["error"],
        "new-cap": ["error"],
        "new-parens": ["error"],
        "no-array-constructor": ["error"],
        "no-lonely-if": ["error"],
        "no-mixed-operators": ["error"],
        "no-nested-ternary": ["error"],
        "no-new-object": ["error"],
        "no-spaced-func": ["error"],
        "no-trailing-spaces": ["error"],
        "no-unneeded-ternary": ["error"],
        "no-whitespace-before-property": ["error"],
        "require-jsdoc": ["error", {
            "require": {
                "FunctionDeclaration": true,
                "MethodDefinition": true,
                "ClassDeclaration": false
            }
        }],
        "valid-jsdoc": ["error", {
            "requireParamDescription": false,
            "requireReturnDescription": false,
            "requireReturn": false
        }],
        "semi": ["error", "always"],
        "semi-spacing": ["error"],
        "space-before-blocks": ["error"],
        "space-before-function-paren": ["error"],
        "space-in-parens": ["error"],
        "space-infix-ops": ["error"],
        "space-unary-ops": ["error"],
        "unicode-bom": ["error"],

        "arrow-spacing": ["error"],
        "no-duplicate-imports": ["error"],
        "no-useless-computed-key": ["error"],
        "no-useless-constructor": ["error"],
        "no-useless-rename": ["error"],
        "no-var": ["error"],
        "prefer-rest-params": ["error"],
        "rest-spread-spacing": ["error"],

        "flowtype/require-parameter-type": ["error"],
        "flowtype/require-return-type": ["error", "always", {"annotateUndefined": "never"}],
        "flowtype/space-after-type-colon": ["error", "always"],
        "flowtype/space-before-type-colon": ["error", "never"],
        "flowtype/type-id-match": ["error", "^([A-Z][a-z0-9]+)+Type$"]
    }
};
```

**What did you do? Please include the actual source code causing the issue.**
Linting the following file:
```javascript
class Foo {
	constructor (context = {} : Object) {

	}
}
```

Via the following command:
```
./node_modules/.bin/eslint <Filename>
```

**What did you expect to happen?**
A message showing me detected errors or parsing errors.

**What actually happened? Please include the actual, raw output from ESLint.**
ESlint crashed.

```
Cannot read property '0' of undefined
TypeError: Cannot read property '0' of undefined
    at SourceCode.api.getFirstToken (/media/p/blg/api-definition/node_modules/eslint/lib/token-store.js:153:40)
    at /media/p/blg/api-definition/node_modules/eslint-plugin-flowtype/dist/rules/spaceAfterTypeColon.js:26:40
    at arrayEach (/media/p/blg/api-definition/node_modules/lodash/lodash.js:485:11)
    at Function.forEach (/media/p/blg/api-definition/node_modules/lodash/lodash.js:8847:14)
    at EventEmitter.<anonymous> (/media/p/blg/api-definition/node_modules/eslint-plugin-flowtype/dist/rules/spaceAfterTypeColon.js:21:26)
    at emitOne (events.js:82:20)
    at EventEmitter.emit (events.js:169:7)
    at NodeEventGenerator.enterNode (/media/p/blg/api-definition/node_modules/eslint/lib/util/node-event-generator.js:40:22)
    at CodePathAnalyzer.enterNode (/media/p/blg/api-definition/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:608:23)
    at CommentEventGenerator.enterNode (/media/p/blg/api-definition/node_modules/eslint/lib/util/comment-event-generator.js:97:23)
```

**Comment by danharper** [comment]
It's not a nice error, but I think Babylon is parsing it when it shouldn't.

The type annotation should come before the default parameter: https://flowtype.org/docs/functions.html#es2015-features

Incorrect:

```js
class Foo {
    constructor (context = {} : Object) {

    }
}
```

Correct:

```js
class Bar {
    constructor (context : Object = {}) {

    }
}
```

Babel parses both:

https://babeljs.io/repl/#?evaluate=true&lineWrap=false&presets=es2015%2Creact%2Cstage-3&experimental=false&loose=false&spec=false&code=class%20Foo%20%7B%0A%20%20%20%20constructor%20(context%20%3D%20%7B%7D%20%3A%20Object)%20%7B%0A%0A%20%20%20%20%7D%0A%7D%0A%0Aclass%20Bar%20%7B%0A%20%20%20%20constructor%20(context%20%3A%20Object%20%3D%20%7B%7D)%20%7B%0A%0A%20%20%20%20%7D%0A%7D%0A&playground=true&stage=0

Flow errors on the first example:

https://flowtype.org/try/#0PQKgBAAgZgNg9gdzCYAoVBjGBDAzrsAMTjjAG9UwqwM4A7XAFwCcBXDRuZsAClrsYBTAB6MwAXnIBfMAC4wAeQBGAK0EcAlOXTUwU1Psw58YAELZuFXfyZsOXXvyGi5i1erGSyUrRUrV9QyA

**Comment by kevinSuttle** [comment]
I have something similar:

```js
export type LabelPropTypes = {
  value:string,
  type?: 'button' | 'toggle' | 'error' | 'input' | 'menu',
  htmlFor?:string,
  children?: Array<HTMLElement>,
}
```

No linting errors.

**Comment by vhf** [comment]
I ran into the exact same issue.

Minimal repro in my case confirms @danharper answer:

```js
function x(a: number = 1) {
  return a;
}
// following crashes eslint because the syntax is not correct
function x(a = 1: number) {
  return a;
}
```

```
// package.json
    "babel-eslint": "^6.1.2",
    "eslint": "^3.2.2",
    "eslint-plugin-flowtype": "^2.6.1",
    

// stack trace
> eslint .
Cannot read property '0' of undefined
TypeError: Cannot read property '0' of undefined
    at SourceCode.api.getTokenBefore (/some-path/node_modules/eslint/lib/token-store.js:89:40)
    at /some-path/node_modules/eslint-plugin-flowtype/dist/rules/spaceBeforeTypeColon.js:26:50
    at arrayEach (/some-path/node_modules/lodash/lodash.js:485:11)
    at Function.forEach (/some-path/node_modules/lodash/lodash.js:8882:14)
    at EventEmitter.<anonymous> (/some-path/node_modules/eslint-plugin-flowtype/dist/rules/spaceBeforeTypeColon.js:21:26)
    at emitOne (events.js:101:20)
    at EventEmitter.emit (events.js:188:7)
    at NodeEventGenerator.enterNode (/some-path/node_modules/eslint/lib/util/node-event-generator.js:40:22)
    at CodePathAnalyzer.enterNode (/some-path/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:608:23)
    at CommentEventGenerator.enterNode (/some-path/node_modules/eslint/lib/util/comment-event-generator.js:97:23)
    at Controller.traverser.traverse.enter (/some-path/node_modules/eslint/lib/eslint.js:903:36)
```

**Comment by danharper** [comment]
> I have something similar:
> 
> export type LabelPropTypes = {
>   value:string,
>   type?: 'button' | 'toggle' | 'error' | 'input' | 'menu',
>   htmlFor?:string,
>   children?: Array<HTMLElement>,
> }
> No linting errors.

Yeah it currently doesn't enforce rules on a TypeAlias/ObjectTypeAnnotation. I might look into that tonight.
